--- a/configure.ac
+++ b/configure.ac
@@ -343,11 +343,11 @@
 AC_SUBST([OPENLSWS_BSSL])
 
 if test "$OPENLSWS_BSSL" = no ; then
-    usedynossl=no
-    CPPFLAGS=" -I../../ssl/include $CPPFLAGS -I../../src -I../../include -I/usr/local/include -I/usr/local -I/usr -I/usr/pkg "
-    echo "Will build latest stable openssl libraries for you, this may take several minutes ..."
-    OSSL=`. $srcdir/dlossl.sh`
-    echo "Finsihed building openssl."
+    usedynossl=yes
+    CPPFLAGS=" $CPPFLAGS -I../../src -I../../include -I/usr/local/include -I/usr/local -I/usr -I/usr/pkg "
+    echo "Skipping bundled openssl build, using system openssl..."
     AC_DEFINE([USE_OPENSSL], [1], [will use openssl as ssl])
 else
